src/social/api.ts

```diff
@@ -1,124 +1,222 @@
-import { Post } from '../models/Post';
+import { PublicPost, Post } from '../models/Post';
+import { Feed } from '../models/Feed';
 import { ImageData } from '../models/ImageData';
-import * as Swarm from '../Swarm';
-import { serialize, deserialize } from './serialization';
-import { uploadPost } from '../PostUpload';
-import { ModelHelper } from '../models/ModelHelper';
-import { MockModelHelper } from '../models/__mocks__/ModelHelper';
-import { Debug } from '../Debug';
+import * as Swarm from '../swarm/Swarm';
 
 type PostCommandType = 'update' | 'remove';
 
-const PostCommandProtocolVersion = 1;
-
-interface PostCommand {
-    protocolVersion: number;
+export const PostCommandProtocolVersion = 1;
 
+/**
+ * This is used to identify a command, so this must be unique in a {@link PostCommandLog}
+ */
+interface PostCommandId {
+    /**
+     * The Lamport-timestamp of the command
+     */
     timestamp: number;
-    parentTimestamp: number;
+    /**
+     * Identifies the user or device where the command is originated
+     */
+    source: string;
+}
 
+const defaultPostCommandId = {
+    timestamp: 0,
+    source: '',
+};
+
+/**
+ * Encapsulates the updates of posts
+ */
+export interface PostCommand {
+    /**
+     * Used for versioning
+     */
+    protocolVersion: number;
+
+    /**
+     * Unique identifier of the command
+     */
+    id: PostCommandId;
+    /**
+     * If the command updates or removes an existing post then this is set
+     * to reference the previous command
+     */
+    parentId: PostCommandId;
+
+    /**
+     * Can be update or remove
+     */
     type: PostCommandType;
+    /**
+     * The actual post content
+     */
     post: Post;
-    source: string;
 
+    /**
+     * If the command is stored on Swarm, this is set to the {@link Swarm.Epoch}
+     * of the update
+     */
     epoch?: Swarm.Epoch;
```

Comment:
just a note for the future: 
to generalize this, we could pick some other name instead of epoch/previousEpoch, and using a union type when we want to support other platforms with different concepts

Reply:
Yes, that was my idea too, just didn't wanted to generalise it now. Actually if we wouldn't use epochs just (bzz) links, it would be much flexible and less tied to Swarm. On the other hand we would lose the ability to sort by epoch, then we would have to provide a sorter function on the interface, so I just left it as is for now.